### DISCUSSION TITLE
enable biome extension to "monitor or alike" the workspace

### DISCUSSION THREAD

**Discussion by emedeirox** [biomejs/biome-vscode#121] [Ideas]
"PROBLEMS" Tab only show the problems of the files that are opened.
Since biomejs is blazing fast it could monitor or check at opening the workspace.

with typescript I can accomplish adding this to JSON settings: 

```json
"typescript.tsserver.experimental.enableProjectDiagnostics": true
```

![image](https://github.com/biomejs/biome-vscode/assets/13124138/f0bd1f80-b427-4422-b3c2-f1b4082af0e2)

**Comment by hornta** [comment]
For me at least, this will make it easier to adopt Biome because when we switched from eslint to biome we got around 200 "errors" but only a handful of them was put into the Problems tab so I had to continously run biome lint and look through the output for each file to open and fix.

**Comment by Hahlh** [comment]
This would indeed be nice to have.

What's the current workaround to get a similar behavior?

**Comment by evheniydan** [comment]
Any updates on this? ESLint has [this eslint.lintTask.enable](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint#using-the-extension-with-vs-codes-task-running) which enables whole workspace linting (not only opened files). Any alternatives/workarounds in Biome?

**Reply by nhedger** [reply]
This hasn't changed, and there are no immediate plans to implement it.

**Comment by tomkel** [comment]
Here's my vscode task to lint the whole project and parse the output into the problems pane, adapted from [here](https://github.com/WSDOT-GIS/wsdot-mp-map/blob/350ec7b13277afcf268eb61df777cf6fb67f743d/.vscode/tasks.json#L107)

```
    {
      "label": "biome lint whole project",
      "type": "process",
      "command": "biome",
      "args": ["lint", "app", "--reporter=github"],
      "presentation": {
        "reveal": "always",
        "panel": "new"
      },
      "problemMatcher": {
        "owner": "biome",
        "source": "biome",
        "pattern": [
          {
            "regexp": "^::(?<severity>\\S+)\\s+title=(?<title>.+),file=(?<file>.+),line=(?<line>\\d+),endLine=(?<endLine>\\d+),col=(?<col>\\d+),endColumn=(?<endColumn>\\d+)::(?<message>.+)$",
            "kind": "location",
            "severity": 1,
            "code": 2,
            "file": 3,
            "line": 4,
            "endLine": 5,
            "column": 6,
            "endColumn": 7,
            "message": 8
          }
        ]
      }
    },

```

As far as the Language Server, it could probably be done with [`workspace/diagnostic`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_diagnostic)

**Reply by planchon** [reply]
works perfectly thank you

Would be great to have it directly in the VSCode extension :)
